notes_app/notes/api.py

```diff
@@ -160,9 +166,37 @@ def login(self, request, **kwargs):
                 'error': 'Wrong username or password'
             }, HttpUnauthorized)
 
+    def register(self, request, **kwargs):
+        self.method_check(request, allowed=['post'])
+        credentials = self.deserialize(
+            request, request.body,
+            format=request.META.get('CONTENT_TYPE', 'application/json')
+        )
+        username = credentials.get('username')
+        password = credentials.get('password')
+        if not username:
+            return self.create_response(request, {
+                'success': False,
+                'error': 'No username in the request'
+            }, HttpBadRequest)
+        if not password:
+            return self.create_response(request, {
+                'success': False,
```

Comment:
maybe it's better to look not success http response code instead of using `success` var. Sometimes it could be 500 internal service error response without your data.